core/src/test/scala/other/kafka/StressTestLog.scala

```diff
@@ -123,7 +123,8 @@ object StressTestLog {
   class WriterThread(val log: UnifiedLog) extends WorkerThread with LogProgress {
     override def work(): Unit = {
       val logAppendInfo = log.appendAsLeader(TestUtils.singletonRecords(currentOffset.toString.getBytes), 0)
-      require(logAppendInfo.firstOffset.forall(_.messageOffset == currentOffset) && logAppendInfo.lastOffset == currentOffset)
+      require((!logAppendInfo.firstOffset.isPresent || logAppendInfo.firstOffset.get().messageOffset == currentOffset)
```

Comment:
It seems that `logAppendInfo.firstOffset.isEmpty` is simpler than `!logAppendInfo.firstOffset.isPresent`.

Reply:
Unfortunately isEmpty doesn't exist in Java 8. When we drop support support for Java 8 (AK 4.0), we'll be able to use it.

Reply:
Right, `isEmpty` is not present in JDK8. That is why we do not use that in several other places also even though it is simpler.